### DISCUSSION TITLE
Melhoria no Page Dynamic Table

### DISCUSSION THREAD

**Discussion by leandrodefaria** [po-ui/po-angular#1153] [Ideias]
Pessoal,
Conforme  #1146, o Page Dynamic Table não carrega os objetos caso o json de retorno não esteja de acordo com a estrutura abaixo:

`"items": [ { "id": "0148093543698", "name": "Ronaldo Nascimento Filho", "birthdate": "1980-11-01T00:00:00-00:00", "genre": "male", }]`

Estou consumindo uma api de um sistema legado e o retorno é desta forma:

`[ { "id": 3, "name": "teste", "corporate_name": null, "cgc": "2525569000158", "phone1": "31258887878"}]`

A questão é que nem todas as api's trabalham conforme o po ui espera, então uma sugestão seria flexibilização quanto ao retorno. 

Obs: Para essa situação, tive que implementar "Page List" combinado com o "Table", retirando o `response.items` e deixando apenas `response`

`getClientList(): void {
    this.ClientListService.getClientList().subscribe(response => {
      this.clientList = response;
    });`

**Comment by alinelariguet** [comment]
Boa tarde @leandrodefaria,
Este comportamento está documentado no nosso portal no [Guia de implementação de Api](https://po-ui.io/guides/api).
Estamos à disposição em caso de qualquer dúvida.

**Comment by leandrodefaria** [comment]
Bom dia @alinelariguet 

Dei uma olhada nessa documentação, mas a minha duvida se existe alguma forma de alterar a estrutura do retorno para um coleção (lista de itens). Ex: Caso seja necessário consumir uma API de terceiros, já pronta e já utilizada para integrar outros sistemas com uma estrutura diferente para retorno, com o Page Dynamic Table não conseguimos usar?
